test/util/test_case.hpp

```diff
@@ -0,0 +1,95 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <utility>
+
+#include "ngraph/function.hpp"
+#include "ngraph/ngraph.hpp"
+
+namespace ngraph
+{
+    namespace test
+    {
+        class NgraphTestCase
+        {
+        public:
+            NgraphTestCase(std::shared_ptr<Function> function, const std::string& backend_name)
+                : m_function(std::move(function))
```

Comment:
In general moving shared pointers is not a common practice. You can change the first parameter of this constructor to `std::shared_ptr&` and remove the `std::move` instead.